var/spack/repos/builtin/packages/opengl/package.py

```diff
@@ -0,0 +1,58 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Opengl(Package):
+    """Placeholder for external OpenGL libraries from hardware vendors"""
+
+    homepage = "https://www.opengl.org/"
+    url      = "https://www.opengl.org/"
+
+    version('3.2', 'N/A')
+
+    provides('gl@:4.5', when='@:4.5')
+    provides('gl@:4.4', when='@:4.4')
+    provides('gl@:4.3', when='@:4.3')
```

Comment:
My guess is that the `:` should go "on the other side" of the `when` clause. Right now this says that for example `opengl@3.2` provides `gl@4.3`, `gl@4.4`, and `gl@4.5` because `opengl@3.2` satisfies all of these constraints.

I would have expected something like:

```
provides('gl@:4.5', when='@4.5:') #opengl 4.5 or higher provides gl 4.5
provides('gl@:4.4', when='@4.4:') ...
```

Or am I missing something? I should say I'm not familiar with opengl versioning specifically but for example it looks like you added the constraints I would expect for `mesa`.